### ISSUE TITLE
annotation rule triggers false positive when annotation with parameters is on parameter type

### ISSUE THREAD

**Issue by matejdro** [pinterest/ktlint#1813]
## Observed Behavior

Following code:

```kotlin
@AnnotationA
fun Demo(arg: @AnnotationB(10) String) {
}
```

produces

`Annotation with parameter(s) should be placed on a separate line prior to the annotated construct (annotation)`

error message. Annotations are already on two separate lines, so it's not clear how to even resolve this. Likely false positive.

## Expected Behavior

Above code should not trigger any rules

## Your Environment

* Version of ktlint used: 0.48.2

**Comment by paul-dingemans** [comment]
With new `ktlint_official` code style, the original code gets formatted as:
```
@AnnotationA
fun Demo(
    arg:
        @AnnotationB(10)
        String,
) {
}
```

**Comment by matejdro** [comment]
Isn't that code style very unreadable? It inserts newlines for no apparent reason.

**Comment by paul-dingemans** [comment]
Yeah, it is not the one thing that I am most proud of. But it does follow the coding conventions (https://kotlinlang.org/docs/coding-conventions.html#annotations) and is in sync with the way how annotations above functions and classes are formatted. So there is an apparent reason for inserting the newlines.

With a 'simple' example like yours, it is very debateable whether code readability has increased/decreased. But, if you would have more annotations on the type, it makes more sense to insert newlines.

> fun demo(arg: @AnnotationA @Annotation @AnnotationC String) {}
